Cubical/Categories/Site/Sheafification/UniversalProperty.agda

```diff
@@ -0,0 +1,183 @@
+{-# OPTIONS --safe #-}
+module Cubical.Categories.Site.Sheafification.UniversalProperty where
+
+-- We prove the universal property of the sheafification,
+-- exhibiting it as a left adjoint to the forgetful functor.
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Structure
+open import Cubical.Foundations.Function using (_$_)
+open import Cubical.Foundations.Equiv
+
+open import Cubical.Data.Sigma
+
+open import Cubical.Categories.Category
+open import Cubical.Categories.Presheaf
+open import Cubical.Categories.Functor
+open import Cubical.Categories.NaturalTransformation
+open import Cubical.Categories.Instances.Sets
+open import Cubical.Categories.Constructions.FullSubcategory
+
+open import Cubical.Categories.Site.Cover
+open import Cubical.Categories.Site.Coverage
+open import Cubical.Categories.Site.Sheaf
+
+open import Cubical.Categories.Site.Sheafification.Base
+open import Cubical.Categories.Site.Sheafification.ElimProp
+
+
+module UniversalProperty
+  {ℓ ℓ' ℓcov ℓpat : Level}
+  {C : Category ℓ ℓ'}
+  (J : Coverage C ℓcov ℓpat)
+  where
+
+  -- We assume 'P' to have this level to ensure that 'F' has the same level as 'P'.
```

Comment:
Ultimately it would of course be nice to know that sheafification has the expected universal property also when it raises universe levels, and it seems the only obstacle to this is that you have to talk about natural transformations between presheaves of different levels? Perhaps this comment could be elaborate to make clear that this is mathematically an unnecessary restriction.

Reply:
Done.